NEWS.md

```diff
@@ -418,6 +413,14 @@ This section lists changes that do not have deprecation warnings.
   * `findn(x::AbstractArray)` has been deprecated in favor of `findall(!iszero, x)`, which
     now returns cartesian indices for multidimensional arrays (see below, [#25532]).
 
+  * Broadcasting operations are no longer fused into a single operation by Julia's parser.
+    Instead, a lazy `Broadcasted` wrapper is created, and the parser will call
+    `copy(bc::Broadcasted)` or `copyto!(dest, bc::Broadcasted)`
```

Comment:
Ah, yes, that's true.  I still want to highlight `copy` and `copyto!` as those are the methods to specialize.  Let's just remove that "parser will call" to make it technically correct.

Reply:
Is there a reason why the parser doesn't lower to these calls directly?

Reply:
Good question.  There are several reasons here, and they're different for in-place (`broadcast!` / `.=`) vs. out-of-place broadcasting.

For in-place broadcasts, we use `materialize!` for two purposes: (1) the outermost `broadcasted` might not return a `Broadcasted` object, but we need to support broadcasting into the destination (e.g., `copyto!(rand(3,3), 1:3)` doesn't broadcast) and we want to use the same code-paths here.  It solves both problems at once by wrapping the non-`Broadcasted` object `x` in a `Broadcasted(identity, (x,))`.  (2) The `Broadcasted` argument wants to know about its axes, but those are dependent upon the axes of the destination. It explicitly sets the axes of the `Broadcasted` object to those of the destination, and then those are checked to ensure they are a compatible shape with `instantiate`.

For out-of-place broadcasts, `materialize` also serves two purposes: (1) for `Broadcasted` arguments, it computes and caches the axes of the outermost `Broadcasted`, recursively checking the axes of its arguments by `instantiate`-ing them.  Now, we may be able to with not storing the `axes` since `axes` should theoretically only be called once and it can be computed on demand, but I like the symmetry with in-place broadcasting and it seems strange to throw a `DimensionError` from a call to `axes`.  (2) for all other arguments it defaults to `identity` but can serve as a hook for custom types to do similar sorts of specializations decoupled from `copy`.

I'm also aiming for some semblance of symmetry between the two cases.

Now there ~~are two~~ is one caveat here: ~~(1) A wart that I really dislike is [`broadcast`](https://github.com/JuliaLang/julia/pull/26891/files#diff-8e941cd1878722cc12a54e01537b8e18R692) itself is hard-coded to call `copy(instantiate(broadcasted(…)))` instead of `materialize`.  For reasons beyond my comprehension, `materialize` sometimes failed to inline even though it is explicitly marked as `@inline`.~~ (Fixed in 8a2d88a — this hack no longer necessary).  (2) The expression `A .= x` still lowers to `broadcast!(identity, A, x)`.  If otherwise unspecialized, it will fall back to the default implementation, which constructs the appropriate `Broadcasted` object and then goes through the above machinery as expected.  We may want to change this purely to be able to simplify our description of this parser transformation.